src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1330,16 +1330,16 @@ public static bool IsDedicatedAdminConnection(ConnectionDetails connectionDetail
         /// Build a connection string from a connection details instance
         /// </summary>
         /// <param name="connectionDetails"></param>
-        public static string BuildConnectionString(ConnectionDetails connectionDetails)
+        public static string BuildConnectionString(ConnectionDetails connectionDetails, bool forceDisablePooling = true)
         {

```

Comment:
do you not having any rules about breaking changes and preserving existing contracts?

Reply:
probably this method should be `internal` but we really don't have API-type scenarios for these services, so typically if the project compiles than it is backward compatible with all its consumers.  Though this isn't always the case so some caution is warranted.

Reply:
Not in here, since this is internal only. This is however a non-breaking API change.